### DISCUSSION TITLE
Is it possible to get each player's position in each tick?

### DISCUSSION THREAD

**Discussion by DeejayArroba** [saul/demofile#204] [Questions] [answered]
Haven't seen an example of it, just wondering.

**Comment by saul** [comment, marked as answer]
⚠️ **Edit 2018-01-05**: Note that it is better to use `demoFile.players[i].position` now.

---

Yes:

``` js
let ent = demoFile.entities.entities[entIndex];
let xy = ent.getProp('DT_CSNonLocalPlayerExclusive', 'm_vecOrigin');

let pos = {
  x: xy.x,
  y: xy.y,
  z: ent.getProp('DT_CSNonLocalPlayerExclusive', 'm_vecOrigin[2]')
}
```

Where entIndex is the entity index of the player in the range `[1..maxClients)`. Note that the userinfo for any disconnected player will be null, so it'll be best to check that before trying to grab the properties.

In CSGO, position XY is stored in "m_vecOrigin", and the Z in "m_vecOrigin[2]". Hope that makes sense.

**Comment by DeejayArroba** [comment]
Ok, thanks for the fast response :)

**Comment by saul** [comment]
No worries - just added some extra info as to what entIndex should be

**Comment by DeejayArroba** [comment]
Hey, I've only gotten to try this now and I have no idea of how to implement it. Do I put this in the update event or tickend, etc? I've tried on update but it only gives me the position once for one player. I want to get all the player's positions every tick.

**Comment by saul** [comment]
If you only care about when the position changes, listen on `demoFile.entities.on('change')`. If you want to do it every tick, use `tickend`.

**Comment by DeejayArroba** [comment]
Alright, I got it working, but the entity change event only gives me changes for player entities (class id 35). Is there a way to also get the position of grenades (and other entities) as well?

**Comment by saul** [comment]
You should see 'change' fire for all entities (you are using GOTV demos, right?), however the position of non-player entities is likely to be in the DT_BaseEntity table as opposed to DT_CSNonLocalPlayerExclusive.

**Comment by DeejayArroba** [comment]
Oh, didn't even notice that. Is there a list of all the tables and properties that are available?

**Comment by saul** [comment]
Try here: https://github.com/L33T/CSGO-Reflection

**Comment by DeejayArroba** [comment]
Nice, thanks.
